cmake/FindMKL.cmake

```diff
@@ -0,0 +1,414 @@
+# - Find INTEL MKL library
```

Comment:
I assume this is all original code but the file is missing a copyright notice.

Rather surprising that there wasn't a cmake macro already available. Of course this way we get exactly what we need.

Reply:
This was from pytorch, but I am having to rewrite it from scratch now, so disregard.  We only want to link to `libmkl_rt.so`.  And indeed no "official" cmake check exists for MKL, although many people asked on the forums for Intel to make one.

Reply:
I added a copyright to the new custom MKL check.